### DISCUSSION TITLE
New VDH beta 9.1.0.33a1 for Firefox released

### DISCUSSION THREAD

**Discussion by GrampaWildWilly** [aclap-dev/video-downloadhelper#1070] [Help and Q&A]
### Your question

Paul is a bit . . . let's say forgetful . . . about the process of officially announcing things.  I decided to check for updates to VDH, something I do maybe once or twice a day.  I just discovered we have a new beta release for Firefox.

![#01](https://github.com/user-attachments/assets/d2f59a5d-8ece-4005-9f42-bd0249207f0d)

I don't know about Chrome & Edge.  I leave it to my fellow users of those browsers to do their own discovering.

The biggest thing in this release is something we users have been begging for for years.

![#02](https://github.com/user-attachments/assets/122febac-f2a9-4b69-ad93-4d7510d920d5)

Finally!  A history activity log!  It's still got some rough edges to smooth down.  The tooltip that hovers up looks like some internal programmer speak.  Paul will get to the fit & finish eventually, I'm sure.  You need to click that new control in that image.  That will take you to this page.

![#03](https://github.com/user-attachments/assets/7b08c712-b55d-409f-9f86-32765f3aa332)

Again, there's some fit & finish that needs to be cleaned up.  The default state of the history log is that it is disabled.  Not functioning.  Not activated.  You have to activate it.  You don't suddenly get history without knowing that you are getting it.  Nobody can say gee I didn't know VDH was keeping history for me.  VDH will keep no history unless you tell it to do so.  Then you will have no excuse to not know it's happening.  You activate history by clicking the green button in that latest image.  When you click the green button, you get this:

![#04](https://github.com/user-attachments/assets/2383213b-c37d-4080-a712-ab38d9aac0a8)

The controls, despite their not-quite-ready-for-prime-time labels, should be self-explanatory.  Over time, you will see entries appearing on this page.  I leave it as an exercise for you to discover what that looks like.

You tell VDH how long to keep this history.  The default retention period is 30 days.  There's a new setting for that in the `More setiings` section of VDH settings.

![#05](https://github.com/user-attachments/assets/daef7aea-7aa7-4872-91ac-d5874a138f0f)

The initial retention period out of the box is 30 days.  You can adjust this to your liking.  I think I'm going to set mine to 2.  You do what's right for you.

I suppose we'll discover what else is new as we use this new release.  The welcome page you get when you install this release does mention the new history log.  It also claims a number of web sites reported here as problematic have been addressed.  Plus supposedly downloads have been sped up.  I guess we'll see.  Mine have always been pretty fast so I'm not setting my expectations too high.  But maybe Paul will give us some details on that score & any other solved problems he would care to brag about.

**Comment by GrampaWildWilly** [comment]
The new `Copy information to clipboard` looks like this:

```
version: 9.1.0.33a1
target: mozilla
channel: beta
lang: en-US
coapp: {"found":true,"path":"C:\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"<removed>"}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:128.0) Gecko/20100101 Firefox/128.0
{
    "valid_license_message_has_been_displayed": true,
    "used_history_button": true,
    "download_directory": "M:\\VDH Testing",
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive",
            "__serializer_value": "M:\\PH"
        }
    },
    "concurrent_downloads_max": 32,
    "show_success_notification_for_icognito": true,
    "theme": "system",
    "view_options": {
        "all_tabs": true,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": true,
        "show_button_convert_local": true,
        "hide_downloaded": false
    },
    "default_action": "download_as",
    "use_wide_ui": true,
    "use_sidebar": false,
    "open_count_store": 10,
    "successfull_dl": 84,
    "record_download_history": true,
    "history_limit_in_days": 2
}
```

The last 2 lines there are new.  There's also a new line about history higher up.

**Comment by paulrouget** [comment]
This hasn't been announced because I found a last minute bug :)

Official announcement: https://github.com/aclap-dev/video-downloadhelper/discussions/1071

**Reply by GrampaWildWilly** [reply]
Indeed.  I notice that you've fixed the fit & finish I mentioned above.  The tooltip & the buttons on the history page are now what they're supposed to be.
